packages/launch/src/focused-launch/summary/index.tsx

```diff
@@ -269,37 +269,27 @@ const PlanStep: React.FunctionComponent< PlanStepProps > = ( {
 	hasPaidDomain = false,
 	selectedPaidDomain = false,
 } ) => {
-	const { setPlan, unsetPlan } = useDispatch( LAUNCH_STORE );
+	const { setPlan } = useDispatch( LAUNCH_STORE );
 
 	const selectedPlan = useSelect( ( select ) => select( LAUNCH_STORE ).getSelectedPlan() );
 
-	const onceSelectedPaidPlan = useSelect( ( select ) => select( LAUNCH_STORE ).getPaidPlan() );
+	const selectedPaidPlan = useSelect( ( select ) => select( LAUNCH_STORE ).getPaidPlan() );
 
 	const { defaultPaidPlan, defaultFreePlan, planPrices } = usePlans();
 
-	const [ nonDefaultPaidPlan, setNonDefaultPaidPlan ] = React.useState< Plan | undefined >();
+	// persist selected plan if it's paid in order to keep displaying it in the
+	const onceSelectedPaidPlan = React.useRef( selectedPaidPlan );
 
 	const isPlanSelected = ( plan: Plan ) => plan && plan.storeSlug === selectedPlan?.storeSlug;
 
-	const sitePlan = useSite().sitePlan;
+	const { sitePlan } = useSite();
 
-	React.useEffect( () => {
-		// To keep the launch store state valid,
-		// unselect the free plan if the user selected a paid domain.
-		// free plans don't support paid domains.
-		if ( selectedPaidDomain && selectedPlan && selectedPlan.isFree ) {
-			unsetPlan();
-		}
-	}, [ selectedPaidDomain, selectedPlan, unsetPlan ] );
```

Comment:
We're already doing this in https://github.com/Automattic/wp-calypso/blob/trunk/packages/launch/src/hooks/use-domain-selection.ts#L20-L22

Reviewed as part of #47993